src/jsutils/symbolSupport.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+/**
+ * A function that can either simply determine if `Symbol`s are allowed as
+ * well as, optionally, determine whether or not a property of symbol
+ * exists.
+ *
+ * If the name of a specific `Symbol` property is supplied, the resulting
+ * value will only be true if property is mapped to an instance of `Symbol`
+ * such as `.toStringTag`, `.iterator`, `.species`, `.isConcatSpreadable`,
+ * etc...
+ *
+ * @method hasSymbolSupport
+ *
+ * @param {string}  specificSymbol an optional name of a property on the
+ * `Symbol` class itself that statically refers to a predefined `Symbol`. If
+ * properly specified and the value is set, then true will be returned.
+ * @return {bool} true if `Symbol` is both defined and a function. Optionally
+ * true if the `Symbol` class is true, a predefined symbol name such as
+ * `toStringTag` is set on the `Symbol` class and it maps to an instance of
+ * `Symbol`. False in all other cases
+ */
+export function hasSymbolSupport(specificSymbol?: string): boolean {
+  const hasSymbols: boolean = typeof Symbol === 'function';
+
+  if (!hasSymbols) {
+    return false;
+  }
+
+  if (specificSymbol) {
+    // NOTE: The capture of type and string comparison over a few lines is
+    // necessary to appease the lint and flowtype gods.
+    //
+    // ((typeof Symbol[specificSymbol]): string) !== 'symbol' makes lint angry
+    // and typeof Symbol[specificSymbol] !== 'symbol' makes flow angry
+    //
+    // The former thinks everything is too verbose the later thinks I am
+    // comparing Symbol instance rather than the string resulting from a call
+    // to typeof.
+    //
+    // Le sigh....
```

Comment:
I think it's premature generalization to create `hasSymbolSupport` since inside `applyToStringTag` you can write:
```js
if (typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol') {
  Object.defineProperty(...)
}
```
\+ it passes Flow

Reply:
This doesn't work exactly as you've typed but a variant in the PR does what you've asked for. I believe the issue here has some fundamental conflict with using `typeof` with `Symbol`.